src/plugins/mf_priority.cpp

```diff
@@ -101,28 +102,42 @@ static void rec_update_cb (flux_t *h,
                            const flux_msg_t *msg,
                            void *arg)
 {
-    char *uid, *fshare, *bank, *default_bank, *max_jobs;
-
-    if (flux_request_unpack (msg, NULL, "{s:s, s:s, s:s, s:s, s:s}",
-                             "userid", &uid,
-                             "bank", &bank,
-                             "default_bank", &default_bank,
-                             "fairshare", &fshare,
-                             "max_jobs", &max_jobs) < 0) {
+    char *uid, *fshare, *bank, *def_bank, *max_jobs;
+    json_t *data, *jtemp = NULL;
+    int num_data = 0;
+
+    if (flux_request_unpack (msg, NULL, "{s:o}", "data", &data) < 0) {
         flux_log_error (h, "failed to unpack custom_priority.trigger msg");
         goto error;
     }
 
     if (flux_respond (h, msg, NULL) < 0)
         flux_log_error (h, "flux_respond");
 
-    struct bank_info *b;
-    b = &users[std::atoi(uid)][bank];
+    if (data != NULL && json_is_array(data)) {
+        num_data = json_array_size(data);
+    } else {
+        flux_log_error (h, "user data unavailable to be unpacked");
+        goto error;
+    }
 
-    b->fairshare = std::atof (fshare);
-    b->max_jobs = std::atoi (max_jobs);
+    for (int i = 0; i < num_data; i++) {
+       json_t *el = json_array_get(data, i);
 
-    users_def_bank[std::atoi (uid)] = default_bank;
+       uid = (char *) json_string_value (json_object_get (el, "userid"));
+       bank = (char *) json_string_value (json_object_get (el, "bank"));
+       def_bank = (char *) json_string_value (json_object_get (el, "def_bank"));
+       fshare = (char *) json_string_value (json_object_get (el, "fairshare"));
+       max_jobs = (char *) json_string_value (json_object_get (el, "max_jobs"));
```

Comment:
Good to know! I didn't know they worked the same. I pushed up a change to just use a chain of `json_unpack ()` on the various elements in the payload. 

Notice that I used `flux_log_error ()` on a failed unpack, which was the same call I had used before with `flux_request_unpack ()`; I just want to make sure the use of `flux_log_error` is correct here since an `errno` is set on the return condition, i.e `json_unpack ()` returns < 0 on failure, right? Let me know if I am misinterpreting.

Reply:
Actually what I'd suggest is to use [json_unpack_ex()](https://jansson.readthedocs.io/en/2.8/apiref.html#c.json_unpack_ex), and capture any resulting error string in a `json_error_t`. Then use `flux_log (h, LOG_ERR, ...)` and include the `error.text` in the output. Though it is unlikely there would be an unpack error here, the actual text error returned by `json_unpack_ex()` will be really useful if it does happen, rather than just the `errno`.

Reply:
Oops! I'm sorry I didn't use that at first. I think what I've pushed up correctly uses `json_unpack_ex ()`, but let me know if anything looks wrong. I passed `0` for the flags parameter and captured any potential error output into a `json_error_t` object `error`. Then, if an error occurs, I just append the `error.text` output to a `"mf_priority unpack: "` error message prefix, like the following:

```c++
flux_log (h, LOG_ERR, "mf_priority unpack: %s", error.text);
```